fetcher/fedora.go

```diff
@@ -0,0 +1,355 @@
+
+func parseModulesYamlFedora(b []byte) (fedoraModuleInfosPerPackage, error) {
+	modules := fedoraModuleInfosPerPackage{}
+	scanner := bufio.NewScanner(bytes.NewReader(b))
+	var contents []string
+	for scanner.Scan() {
+		str := scanner.Text()
+		switch str {
+		case "---":
+			{
+				contents = []string{}
+			}
+		case "...":
+			{
+				var module FedoraModuleInfo
+				err := yaml.NewDecoder(strings.NewReader(strings.Join(contents, "\n"))).Decode(&module)
+				if err != nil {
+					return nil, xerrors.Errorf("failed to decode module info. err: %w", err)
+				}
+				if module.Version == 2 {
+					modules[module.ConvertToUpdateInfoTitle()] = module
+				}
+			}
+		default:
+			{
+				contents = append(contents, str)
+			}
+		}
+	}
+
+	return modules, nil
+}
+
+func fetchCveIDsFromBugzilla(id string) ([]string, error) {
+	req := fetchRequest{
+		url:           fmt.Sprintf("https://bugzilla.redhat.com/show_bug.cgi?ctype=xml&id=%s", id),
+		logSuppressed: true,
+	}
+	log15.Info("Fetch CVE-ID list from bugzilla.redhat.com", "URL", req.url)
+	body, err := fetchFileWithUA(req)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch CVE-ID list, err: %w", err)
+	}
+
+	var b bugzillaXML
+	if err = xml.Unmarshal(body, &b); err != nil {
+		return nil, xerrors.Errorf("Failed to unmarshal xml. url: %s, err: %w", req.url, err)
+	}
+
+	var reqs []fetchRequest
+	for _, v := range b.Blocked {
+		req := fetchRequest{
+			url:           fmt.Sprintf("https://bugzilla.redhat.com/show_bug.cgi?ctype=xml&id=%s", v),
+			concurrently:  true,
+			logSuppressed: true,
+		}
+		reqs = append(reqs, req)
+	}
+
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch CVE-IDs, err: %w", err)
+	}
+
+	var ids []string
+	for _, result := range results {
+		var b bugzillaXML
+		if err = xml.Unmarshal(result.Body, &b); err != nil {
+			return nil, xerrors.Errorf("Failed to unmarshal xml. url: %s, err: %w", req.url, err)
+		}
+
+		if b.Alias != "" {
+			ids = append(ids, b.Alias)
+		}
+	}
+
+	log15.Info(fmt.Sprintf("%d CVE-IDs fetched", len(ids)))
+	return ids, nil
+}
+
+func extractInfoFromRepoMd(results []FetchResult, rt string, mt mimeType) ([]fetchRequest, error) {
+	var updateInfoReqs []fetchRequest
+	for _, r := range results {
+		var repoMd RepoMd
+		if err := xml.NewDecoder(bytes.NewBuffer(r.Body)).Decode(&repoMd); err != nil {
+			return nil, xerrors.Errorf("Failed to decode repomd of version %s. err: %w", r.Target, err)
+		}
+
+		for _, repo := range repoMd.RepoList {
+			if repo.Type == rt {
+				u, err := url.Parse(r.URL)
+				if err != nil {
+					return nil, xerrors.Errorf("Failed to parse URL in XML. err: %w", err)
+				}
+				u.Path = strings.Replace(u.Path, "repodata/repomd.xml", repo.Location.Href, 1)
+				req := fetchRequest{
+					url:          u.String(),
+					target:       r.Target,
+					mimeType:     mt,
+					concurrently: true,
+				}
+				updateInfoReqs = append(updateInfoReqs, req)
+				break
+			}
```

Comment:
How about this coding style?
```suggestion
			if repo.Type != rt {
				continue
			}
			u, err := url.Parse(r.URL)
			if err != nil {
				return nil, xerrors.Errorf("Failed to parse URL in XML. err: %w", err)
			}
			u.Path = strings.Replace(u.Path, "repodata/repomd.xml", repo.Location.Href, 1)
			req := fetchRequest{
				url:          u.String(),
				target:       r.Target,
				mimeType:     mt,
				concurrently: true,
			}
			updateInfoReqs = append(updateInfoReqs, req)
			break
```